### PR TITLE
Added Fleet Input Plugin

### PR DESCRIPTION
**Pull request by jpweber**
This plugin will gather names of running units from fleet and the sum total of each running unit. It uses the fleet v1 API to gather data.

### CODE REVIEW

File: plugins/inputs/fleet/fleet.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+// Description - Method to provide description of plugin
+func (f *Fleet) Description() string {
+	return "Fleetd Plugin to glather information about container states in fleet cluster"
+}
+
+// SampleConfig output sample config for this plugin
+func (*Fleet) SampleConfig() string {
+	return `
+# Description
```
Comment (sparrc) [inline plugins/inputs/fleet/fleet.go]: `# Description` line is not necessary

File: plugins/inputs/fleet/fleet.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+[[inputs.fleet]]
```
Comment (sparrc) [inline plugins/inputs/fleet/fleet.go]: [[inputs.fleet]] should not be here

File: plugins/inputs/fleet/fleet.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+## Works with Fleet HTTP API
```
Comment (sparrc) [inline plugins/inputs/fleet/fleet.go]: indent these lines by two spaces

File: plugins/inputs/fleet/fleet.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+## Multiple Hosts from which to read Fleet stats:
+	host = ["http://localhost:49153/fleet/v1/state"]
+`
+}
+
+// Gather method to gather stats for telegraf input
+func (f *Fleet) Gather(accumulator telegraf.Accumulator) error {
+	errorChannel := make(chan error, len(f.Hosts))
```
Comment (sparrc) [inline plugins/inputs/fleet/fleet.go]: use `internal.errChan` here

**Comment by sparrc** [comment]
@jpweber I could be off-base here, but it looks as if there are many more stats available, and that it might be useful to append the field names with what it is they are counting. 

so append `_running` to the running counts, and also count "active" & "loaded" counts, as well as summing a "total" count. So the outputted metrics would look something like this:

```
> fleet,host=localhost.local,server=http://fleet.testserver.com:49153/fleet/v1/state redis_running=2i,redis_loaded=2i,redis_active=2i,myapp_running=3i,myapp_loaded=3i,myapp_active=3i,total_running=4i,total_loaded=5i,total_active=5i 1470615664000000000
```

**Comment by jpweber** [comment]
@sparrc Your comment about more stats makes total sense. I can get those worked in as well as the other smaller issues you've pointed out.

**Comment by sparrc** [comment]
another option which might be better would be to _tag_ metrics with the app name (as well as a "total" tag), so it would look like this:

```
> fleet,host=localhost.local,name=redis running=2i,loaded=2i,active=2i 1470615664000000000
> fleet,host=localhost.local,name=myapp running=2i,loaded=2i,active=2i 1470615664000000000
> fleet,host=localhost.local,name=total running=4i,loaded=4i,active=4i 1470615664000000000
```

Only concern is series cardinality: is it common for app names to get into the 1000s?

**Comment by jpweber** [comment]
ok I can give that a shot. I would say its _possible_ to get an app name in the thousands, but not likely. At that point one should be using a different cluster orchestration instead of fleet.

**Comment by russorat** [comment]
per the github repo, Fleet is being deprecated: https://github.com/coreos/fleet

**Comment by glinton** [comment]
Due to the deprecation of fleet and the lack of activity, I'm closing this. Feel free to reopen if there is still a need/want. thanks

**Pull request closed without merging**